specification/quantum/data-plane/Microsoft.Quantum/preview/2019-11-04-preview/quantum.json

```diff
@@ -0,0 +1,552 @@
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Provider id."
+        },
+        "currentAvailability": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Provider availability.",
+          "enum": [
+            "Available",
+            "Degraded",
+            "Unavailable"
+          ],
+          "x-ms-enum" : {
+            "name": "ProviderAvailability",
+            "modelAsString": true
+          }
+        },
+        "targets": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/targetStatus"
+          }
+        }
+      }
+    },
+    "targetStatus": {
+      "description": "Target status.",
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Target id."
+        },
+        "currentAvailability": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Target availability.",
+          "enum": [
+            "Available",
+            "Degraded",
+            "Unavailable"
+          ],
+          "x-ms-enum" : {
+            "name": "TargetAvailability",
+            "modelAsString": true
+          }
+        },
+        "averageQueueTime": {
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true,
+          "description": "Average queue time in seconds."
+        },
+        "statusPage": {
+          "type": "string",
+          "readOnly": true,
+          "description": "A page with detailed status of the provider."
+        }
+      }
+    },
+    "restError": {
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "An identifier for the error. Codes are invariant and are intended to be consumed programmatically."
+        },
+        "message": {
+          "type": "string",
+          "description": "A message describing the error, intended to be suitable for displaying in a user interface."
+        }
+      },
+      "description": "An error response from Azure."
+    }
+  },
+  "parameters": {
+    "subscriptionIdParameter": {
+      "name": "subscriptionId",
+      "description": "The Azure subscription ID. This is a GUID-formatted string (e.g. 00000000-0000-0000-0000-000000000000)",
+      "in": "path",
+      "required": true,
+      "x-ms-parameter-location":"client",
+      "type": "string"
+    },
+    "resourceGroupNameParameter": {
+      "name": "resourceGroupName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location":"client",
+      "description": "Name of an Azure resource group."
+    },
+    "workspaceNameParameter": {
+      "name": "workspaceName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location":"client",
```

Comment:
Same here but even worse, it's a surprise you want to freeze the workspace at the client level

Reply:
Same thing, a user typically works with a single workspace which is why we're defining it at the client level, but let's chat.

Reply:
We talked about this during the review and agreed it would be ok for now. Each workspace will have its own endpoint on the next version of the API.